drivers/modem/modem_iface_uart.c

```diff
@@ -125,6 +125,15 @@ static int modem_iface_uart_read(struct modem_iface *iface,
 	return 0;
 }
 
+static bool mux_is_active(struct modem_iface *iface)
+{
+	bool mux_active =
+		IS_ENABLED(CONFIG_GSM_MUX) &&
+		!strncmp(iface->dev->name, CONFIG_UART_MUX_DEVICE_NAME,
```

Comment:
I do not like that we are doing string compare for every send.
Could we do it something like this instead
```
bool mux_active = IS_ENABLED(CONFIG_GSM_MUX) &&
      iface->dev == device_get_binding(CONFIG_UART_MUX_DEVICE_NAME);
```
the `device_get_binding()` has some optimizations for doing the strcmp() as a last resort and also checking if the device is actually ready for action.

Reply:
I think strncmp might be faster in this particular instance, but how about caching the looked up mux device with a static? Then it'll be instantaneous.

Reply:
If we do not want to call `device_get_binding()` here, we can just check the pointer values like what device_get_binding() is doing. I think `iface->dev->name == CONFIG_UART_MUX_DEVICE_NAME` would be enough here.

Reply:
oh yeah you're right actually - didn't even consider that. That's strictly superior.

Reply:
k done